src/inmanta/protocol/methods_v2.py

```diff
@@ -1546,13 +1546,17 @@ def set_password(username: str, password: str) -> None:
     varkw=True,
 )
 def discovered_resource_create(
-    tid: uuid.UUID, discovered_resource_id: str, **kwargs: object  # bypass the type checking for the values
+    tid: uuid.UUID,
+    discovered_resource_id: str,
+    discovery_resource_id: str | None = None,
```

Comment:
In the same vein as my comment regarding optionality, if accepted there, we need to make it non-optional here as well.